test/stub-test.js

```diff
@@ -1345,6 +1345,24 @@ describe("stub", function () {
             assert.equals(stub(), 23);
             assert.equals(stub({ foo: "bar", bar: "foo" }), 99);
         });
+
+        it("ensure stub uses last matching arguments", function () {
+            var stub = createStub().returns(23);
```

Comment:
Try to avoid using literal values for tests, but label them using function names and variable names. This removes a lot of confusion situations and questions, like "is this `2` value the same `2` value as above?".

I realise that it's not practiced throughout the Sinon.JS codebase much yet, it's something I am hoping to change, so that it'll be easier for people to do more casual contributions.

Using [v4 UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier) values for return values, where we don't care about the exact value has the benefit of making it trivial to search for them in a large codebase, unlike values like `2`, `"username"`, etc.

I've tried applying the advice to the first test.

```javascript
it("ensure stub uses last matching arguments", function () {
    var unmatchedValue = "d3ada6a0-8dac-4136-956d-033b5f23eadf";
    var firstMatchedValue = "68128619-a639-4b32-a4a0-6519165bf301";
    var secondMatchedValue = "4ac2dc8f-3f3f-4648-9838-a2825fd94c9a";
    var expectedArgument = "3e1ed1ec-c377-4432-a33e-3c937f1406d1";
    
    var stub = createStub().returns(unmatchedValue);
    
    stub.withArgs(expectedArgument).returns(firstMatchedValue);
    stub.withArgs(expectedArgument).returns(secondMatchedValue);

     assert.equals(stub(), unmatchedValue);
     assert.equals(stub(expectedArgument), secondMatchedValue);
});
```

What do you think?